persistence/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/persistence/PersistentEntity.scala

```diff
@@ -122,42 +117,37 @@ abstract class PersistentEntity[Command, Event, State] {
    */
   def entityTypeName: String = Logging.simpleName(getClass)
 
+  def initialState: State
+
   /**
    * Abstract method that must be implemented by concrete subclass to define
    * the behavior of the entity. Use [[#newBehaviorBuilder]] to create a mutable builder
    * for defining the behavior.
    */
-  def initialBehavior(snapshotState: Option[State]): Behavior
+  def behavior: Behavior
 
   /**
    * This method is called to notify the entity that the recovery process
    * is finished.
    */
-  def recoveryCompleted(currentBehavior: Behavior): Behavior = currentBehavior
-
-  type EventHandler = PartialFunction[(Event, Behavior), Behavior]
-  type CommandHandler = PartialFunction[(Command, CommandContext, State), Persist[Event]]
-  type ReadOnlyCommandHandler = PartialFunction[(Command, ReadOnlyCommandContext, State), Unit]
+  def recoveryCompleted(state: State): State = state
 
-  object Behavior {
-    def apply(state: State): Behavior =
-      new Behavior(state, PartialFunction.empty, PartialFunction.empty)
+  object Actions {
+    private val empty = new Actions(PartialFunction.empty, PartialFunction.empty)
+    def apply(): Actions = empty
   }
 
   /**
-   * Behavior consists of current state and functions to process incoming commands
-   * and persisted events. `Behavior` is an immutable class.
+   * Actions consists of functions to process incoming commands
+   * and persisted events. `Actions` is an immutable class.
    */
-  class Behavior(
-    val state:          State,
+  class Actions(
     val eventHandler:   EventHandler,
     val commandHandler: CommandHandler
   ) {
 
-    def addCommandHandler(handler: CommandHandler) = {
-      new Behavior(state, eventHandler,
-        commandHandler.orElse(handler))
-    }
+    def addCommandHandler(handler: CommandHandler) =
```

Comment:
It think it's rather common to use same event and command handlers in several states and this "adding" instead of "replacing" makes that easier. E.g. https://github.com/patriknw/lagom/blob/df46423359b135365ca1741da897707c8387421d/persistence/scaladsl/src/test/scala/com/lightbend/lagom/scaladsl/persistence/TestEntity.scala#L91